src/compiler/parser.rs

```diff
@@ -462,7 +477,9 @@ impl ParserType {
     }
 
     fn next_token_without_check(&self) -> SyntaxKind {
-        let current_token = self.scanner().scan();
+        let current_token = self
+            .scanner()
+            .scan(Some(&|message, length| self.scan_error(message, length)));
```

Comment:
Per above comment, this is where instead I'm injecting the `on_error` callback into the scanner when calling `.scan()`